simple_retry/decorators.py

```diff
@@ -62,3 +62,31 @@ def f_retry(*args, **kwargs):
             return (yield from function(*args, **kwargs))
         return f_retry
     return deco_retry
+
+
+def async_retry(Except, retries=5, delay=0, logger=None, level='info', multiple=1):
+    def deco_retry(function):
+
+        @wraps(function)
+        async def f_retry(*args, **kwargs):
+            tries = 1
+            mdelay = delay
+            while tries < retries:
+                try:
+                    return await (function(*args, **kwargs))
+                except Except as e:
+                    msg = '{e}, Retrying {tries} of {retries}'.format(
+                        e=e,
+                        tries=tries,
+                        retries=retries
+                    )
+                    if mdelay:
+                        msg = ' '.join([msg, 'in {} seconds...'.format(mdelay)])
+                    if logger:
+                        getattr(logger, level)(msg)
+                    time.sleep(mdelay)
```

Comment:
Here, you need to use `await asyncio.sleep(mdelay)`

Reply:
Done!

Fix for both `coroutine` and `async` decorators.

https://github.com/nicolasmota/simple_retry/pull/6/commits/1041130360fb910df86c281328147eb9f9616f3c

https://github.com/nicolasmota/simple_retry/pull/6/commits/dc46823ba3165eb0a9016c281ffa012e2e4058bd